packages/nouns-contracts/contracts/client-incentives/Rewards.sol

```diff
@@ -0,0 +1,624 @@
+        IERC20 ethToken;
+        /// @dev admin account able to pause/unpause the contract in case of a quick response is needed
+        address admin;
+        /// @dev tracking rewards balances for clients
+        mapping(uint32 clientId => uint256 balance) _clientBalances;
+    }
+
+    /// @dev This is a ERC-7201 storage location, calculated using:
+    /// @dev keccak256(abi.encode(uint256(keccak256("nouns.rewards")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 public constant RewardsStorageLocation = 0x9a06af3161ac5b0c3de4e6c981ab9d9f60b530386f5eaae00d541393fbecd700;
+
+    function _getRewardsStorage() private pure returns (RewardsStorage storage $) {
+        assembly {
+            $.slot := RewardsStorageLocation
+        }
+    }
+
+    /**
+     * @dev Throws if called by any account other than the owner or admin.
+     */
+    modifier onlyOwnerOrAdmin() {
+        RewardsStorage storage $ = _getRewardsStorage();
+        require(owner() == _msgSender() || $.admin == _msgSender(), 'Caller must be owner or admin');
+        _;
+    }
+
+    /**
+     * @dev This constructor does not have the `initializer` modifier, since it inherits from `NounsClientToken` which
+     * has the `initializer` modifier on its constructor.
+     */
+    constructor(address nounsDAO_, address auctionHouse_) {
+        nounsDAO = INounsDAOLogic(nounsDAO_);
+        auctionHouse = INounsAuctionHouseV2(auctionHouse_);
+    }
+
+    function initialize(
+        address owner,
+        address admin_,
+        address ethToken_,
+        uint32 nextProposalIdToReward_,
+        uint32 nextAuctionIdToReward_,
+        uint32 nextProposalRewardFirstAuctionId_,
+        RewardParams memory rewardParams,
+        address descriptor
+    ) public initializer {
+        RewardsStorage storage $ = _getRewardsStorage();
+
+        super.initialize(owner, descriptor);
+        __Pausable_init_unchained();
+        $.admin = admin_;
+        $.ethToken = IERC20(ethToken_);
+        $.nextProposalIdToReward = nextProposalIdToReward_;
+        $.nextAuctionIdToReward = nextAuctionIdToReward_;
+        $.nextProposalRewardFirstAuctionId = nextProposalRewardFirstAuctionId_;
+        $.params = rewardParams;
+        $.lastProposalRewardsUpdate = uint40(block.timestamp);
+    }
+
+    /**
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     *   PUBLIC WRITE
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     */
+
+    /**
+     * @notice Register a client, mints an NFT and assigns a clientId
+     * @return uint32 the newly assigned clientId
+     */
+    function registerClient(
+        string calldata name,
+        string calldata description
+    ) public override whenNotPaused returns (uint32) {
+        RewardsStorage storage $ = _getRewardsStorage();
+
+        uint32 tokenId = super.registerClient(name, description);
+
+        // Increase the balance by one wei so that the slot is non zero when increased in the future
+        $._clientBalances[tokenId] += 1;
```

Comment:
@davidbrai why is this `+=1` rather than `=1`?

Reply:
@eladmallel because someone may pass a `clientId` before it was registered.
it's not a normal use case, but if it happens, I thought maybe better not to zero the balance
wdyt?

Reply:
I think that makes sense, I think it can be helpful to add another inline comment explaining this, wdyt?

Reply:
@eladmallel yes, but maybe not relevant if we're changing the way we track balances? then no need for the 1 wei hack